app/windows/Import/Stores/Stats.js

```diff
@@ -31,6 +34,33 @@ export class StatStorage {
     this.isLoading = false
     this.wasLoading = false
     this.importing = false
+    this.wasLoadingStats = false
+  }
+
+  check () {
+    // Prepare the promise to check the Peers
+    const pPeers = getPeersWithObjectbyHash(this.hash)
+      .then(peers => {
+        console.log('got peers', peers)
+        this.peersAmount = peers.length
+        this.isLoading = false
+      })
+
+    // Prepare the promise to check the stats
+    const pStats = getObjectStat(this.hash)
+      .then(stats => {
+        console.log('got stats')
+        this.stats = stats
+        this.isLoading = false
```

Comment:
What about concurrency?
Should we have a `isLoading` both here and on line 46, or should we have a `.then` right after the `Promise.all`? What are the pro/cons of this?

Reply:
I agree, must better to handle them in `Promise.all`